### PR TITLE
Add init hook flag to shellenv command

### PR DESCRIPTION
**Pull request by LucilleH**
## Summary
Add init hook flag to shellenv command. This prints out the env variables and the init hooks to stdout, and should replicate the behavior of a `devbox shell` within the context of the current shell.

## How was it tested?
```
devbox shellenv --init-hook
```
And changing the shell by
```
eval "$(devbox shellenv --init-hook)"
```

### CODE REVIEW

File: internal/boxcli/shellenv.go [hunk 1]
```diff
@@ -43,3 +51,12 @@ func shellEnvFunc(cmd *cobra.Command, flags shellEnvCmdFlags) (string, error) {
...
 	return box.PrintEnv()
 }
+
+func shellEnvInitHookFunc(cmd *cobra.Command, flags shellEnvCmdFlags) string {
+	box, err := devbox.Open(flags.config.path, cmd.ErrOrStderr())
+	if err != nil {
+		return ""
+	}
+
+	return box.Config().Shell.InitHook.String()
```
Comment (mikeland86) [inline internal/boxcli/shellenv.go:61]: Integrate with `shellEnvFunc`? and add error handling

**Review by mikeland86 — APPROVED** [review]

File: internal/boxcli/shellenv.go (hunk 1 above)
Comment (mikeland86) [reply internal/boxcli/shellenv.go:61]:
```golang
func shellEnvFunc(cmd *cobra.Command, flags shellEnvCmdFlags) (string, error) {
	box, err := devbox.Open(flags.config.path, cmd.ErrOrStderr())
	if err != nil {
		return "", err
	}

	result := box.PrintEnv()

	if flags.runInitHook {
		box, err := devbox.Open(flags.config.path, cmd.ErrOrStderr())
		if err != nil {
			return ""
		}
		result := result + "\n" + box.Config().Shell.InitHook.String()
	}

	return result
}
```

**Comment by ipince** [comment]
Couple questions/suggestions:

* Should `devbox shell --print-env` then print the init hook as well? or, remove that call altogether
* We could instead print `. .devbox/gen/scripts/.hooks.sh` instead of copying the entire hook here

**Comment by LucilleH** [comment]
> Couple questions/suggestions:
> 
> * Should `devbox shell --print-env` then print the init hook as well? or, remove that call altogether

I think it should? I'm not sure. I know it is being used in direnv

> We could instead print . .devbox/gen/scripts/.hooks.sh instead of copying the entire hook here

Will there be a race condition of `.devbox/gen/scripts/.hooks.sh` not being generated in time?

**Pull request merged**